### PR TITLE
596: Allow short-hand syntax for /label command

### PR DESCRIPTION
**Pull request by lgxbslgx**
Hi all,

This patch adds the short-hand syntax for `/label` command. The short-hand syntax makes the users easily add and remove the labels in one comment.

Some example:
```
Add a label: `/label +labelName`
Remove a label: `/label -labelName`
Mixed: `/label +labelName, -labelName2`
Mixed: `/label labelName, +labelName2, -labelName3`
```

And the test cases are added.

Thanks for taking the time to review.

Best Regards,
-- Guoxiong

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [SKARA-596](https://bugs.openjdk.java.net/browse/SKARA-596): Allow short-hand syntax for /label command

### Reviewers
 * [Erik Joelsson](https://openjdk.java.net/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/skara pull/1255/head:pull/1255` \
`$ git checkout pull/1255`

Update a local copy of the PR: \
`$ git checkout pull/1255` \
`$ git pull https://git.openjdk.java.net/skara pull/1255/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1255`

View PR using the GUI difftool: \
`$ git pr show -t 1255`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/skara/pull/1255.diff">https://git.openjdk.java.net/skara/pull/1255.diff</a>

</details>

### CODE REVIEW

**Review by erikj79 — COMMENTED** [review]
Looks good otherwise.

File: bots/pr/src/main/java/org/openjdk/skara/bots/pr/LabelCommand.java [hunk 1]
```diff
@@ -58,49 +62,109 @@ public void handle(PullRequestBot bot, PullRequest pr, CensusInstance censusInst
...
+                return;
+            }
+
+            addLabels(labelsToAdd, currentLabels, pr, reply);
+            removeLabels(labelsToRemove, currentLabels, pr, reply);
+        }
+    }
+
+    private void printInvalidLabels(List<String> invalidLabels, PullRequestBot bot, PrintWriter reply) {
+        reply.println(""); // Intentionally blank line.
```
[the remark this answers is not in the document]
Comment (lgxbslgx) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/LabelCommand.java:123]: The `@username` will be printed at first. If the invalid labels is more than 2. The output will be:

```
@username The label `labelName` is not a valid label.
The label `labelName` is not a valid label.
```

I would like to adjust it to :

```
@username
The label `labelName` is not a valid label.
The label `labelName` is not a valid label.
```

[the remark this answers is not in the document]
Comment (erikj79) [reply bots/pr/src/main/java/org/openjdk/skara/bots/pr/LabelCommand.java:123]: Ah, makes sense.

**Review by erikj79 — APPROVED** [review]

**Comment by lgxbslgx** [comment]
@erikj79 Thanks for the review. Could I get your help to sponsor this patch?

/integrate

**Pull request closed without merging**